clients/client-go/codegenerator/model/api.go

```diff
@@ -359,8 +359,10 @@ func (entry *APIEntry) generateDirectMethod(apiName string) string {
 
 func (entry *APIEntry) generateSignedURLMethod(apiName string) string {
 	// if no required scopes, no reason to provide a signed url
-	// method, since no auth is required, so unsigned url already works
-	if entry.Scopes.Type == "" {
+	// method, since no auth is required, so unsigned url already works,
+	// except for TestAuthenticateGet, which can be usefully used to test
+	// with signed URLs
+	if entry.Scopes.Type == "" && entry.MethodName != "TestAuthenticateGet" {
```

Comment:
I guess this is ok for now, it feels a little hacky though. I'm not sure of an alternative simple solution that would achieve the same, without requiring more fields in the API definition, which also feels like overkill.

I would still advocate for our official CI tests to use real credentials, and then we wouldn't need to introduce this workaround.

Reply:
I don't think the assumption that an API method without scopes doesn't need a `_SignedURL` method is a valid assumption.  Rather than change it wholesale, I made this exception.  So I suppose we agree that this is a workaround, but differ on what it's working around.  Can we just leave it as a compromise?